meson.build

```diff
@@ -0,0 +1,2304 @@
+# -*- mode: meson -*-
+
+project('systemd', 'c',
+        version : '233',
+        license : 'LGPLv2+',
+        default_options: [
+          'c_std=gnu99',
+          'prefix=/usr',
+          'sysconfdir=/etc',
+          'localstatedir=/var',
+        ],
+        meson_version : '>= 0.39.1',
+       )
+
+# We need the same data in three different formats, ugh!
+# Also, for hysterical reasons, we use different variable
+# names, sometimes. Not all variables are included in every
+# set. Ugh, ugh, ugh!
+conf = configuration_data()
+conf.set_quoted('PACKAGE_STRING',  meson.project_name() + ' ' + meson.project_version())
+conf.set_quoted('PACKAGE_VERSION', meson.project_version())
+
+substs = configuration_data()
+substs.set('PACKAGE_URL',          'https://www.freedesktop.org/wiki/Software/systemd')
+substs.set('PACKAGE_VERSION',      meson.project_version())
+
+m4_defines = []
+
+#####################################################################
+
+if get_option('split-usr')
+  conf.set('HAVE_SPLIT_USR', 1)
+  rootprefixdir = '/'
```

Comment:
Gentoo currently builds with the following options:
```
--enable-split-usr
--with-rootprefix=/usr
```
I realize this config is a bit unorthodox, but I think it would be difficult to change it at this point. Can you maintain the independent rootprefix option for the meson build?